src/store/reducers.js

```diff
@@ -131,14 +152,49 @@ function selectedReducer(INVENTORY_ACTIONS, createBaselineModal, historicalProfi
             return {
                 ...state,
                 selectedSystemIds,
-                rows: newRows.length > 0 ? newRows : state.rows
+                rows: newRows.length > 0 ? newRows : state.rows,
+                columns: newColumns
             };
         },
         [types.SET_SELECTED_SYSTEM_IDS]: (state, action) => {
             return {
                 ...state,
                 selectedSystemIds: action.payload.selectedSystemIds
             };
+        },
+        [types.SELECT_SINGLE_HSP]: (state, action) => {
+            newRows = [ ...state.rows ];
+
+            /*eslint-disable camelcase*/
+            newRows.forEach((row) => {
+                row.selected = false;
+                row.display_selected_hsp = row.display_name;
+                if (action.payload.system_id === row.id) {
+                    row.display_selected_hsp = <React.Fragment>
+                        <div>
+                            { row.display_name }
+                        </div>
+                        <div>
+                            <HistoryIcon className='active-blue margin-right-5-px' />
+                            {
+                                moment.utc(action.payload.captured_date).format('DD MMM YYYY, HH:mm UTC')
+                            }
+                            <CloseIcon
+                                className='pointer active-blue margin-left-5-px'
+                                onClick={ () => deselectHistoricalProfiles() }
+                            />
+                        </div>
+                    </React.Fragment>;
+                }
+            });
+            /*eslint-enable camelcase*/
```

Comment:
I could see moving this logic into a helper file to allow unit testing and keep React/JSX out of the Redux

Reply:
@ehelms I moved this into a helper file and tried to write a unit test for it, but it's failing. https://github.com/RedHatInsights/drift-frontend/pull/336/files#diff-d6788e1e81ad3ae6f1541df0a329f364R48

I skipped it because I'm just not sure how to fix it. Perhaps you can take a quick look?